pytorch_lightning/trainer/lr_finder.py

```diff
@@ -0,0 +1,288 @@
+        to reture the amount of guesswork in picking a good starting learning rate.
+
+        Args:
+            model: Model to do range testing for
+
+            train_dataloader: A Pytorch
+                DataLoader with training samples. If the model has
+                a predefined train_dataloader method this will be skipped.
+
+            min_lr: minimum learning rate to investigate
+
+            max_lr: maximum learning rate to investigate
+
+            num_iters: number of learning rates to test
+
+            mode: search strategy, either 'linear' or 'exponential'. If set to
+                'linear' the learning rate will be searches by linearly increasing
+                after each batch. If set to 'exponential', will increase learning
+                rate logarithmic.
+
+            num_accumulation_steps: number of batches to calculate loss over.
+
+        Example::
+            # Setup model and trainer
+            model = MyModelClass(hparams)
+            trainer = pl.Trainer()
+
+            # Run lr finder
+            LRfinder = trainer.find_lr(model, ...)
+
+            # Inspect results
+            LRfinder.plot()
+            suggested_lr = LRfinder.suggest()
+
+            # Overwhite lr in model config
+            model.hparams.lr = suggested_lr
+
+            # Ready to train with new learning rate
+            trainer.fit(model)
+
+        """
+        # Initialize lr finder object (stores results)
+        lr_finder = _LRFinder(mode, min_lr, max_lr, num_iters)
+
+        # Use special lr logger callback
+        callbacks = self.callbacks
+        self.callbacks = [_LRCallback(num_iters, show_progress_bar=True)]
+
+        # No logging
+        logger = self.logger
+        self.logger = None
+
+        # Max step set to number of iterations
+        max_steps = self.max_steps
+        self.max_steps = num_iters
+
+        # Disable standard progress bar for fit
+        show_progress_bar = self.show_progress_bar
+        self.show_progress_bar = False
+
+        # Accumulation of gradients
+        accumulate_grad_batches = self.accumulate_grad_batches
+        self.accumulate_grad_batches = num_accumulation_steps
+
+        # Configure optimizer and scheduler
+        optimizer, _, _ = self.init_optimizers(model.configure_optimizers())
+        assert len(optimizer) == 1, 'cannot find lr for more than 1 optimizer'
+        configure_optimizers = model.configure_optimizers
+        model.configure_optimizers = lr_finder._get_new_optimizer(optimizer[0])
+
+        # Fit, lr & loss logged in callback
+        self.fit(model)
+
+        # Promt if we stopped early
+        if self.global_step != num_iters:
+            print('LR finder stopped early due to diverging loss.')
+
+        # Transfer results from callback to lr finder object
+        lr_finder.results.update({'lr': self.callbacks[0].lrs,
+                                  'loss': self.callbacks[0].losses})
```

Comment:
regarding the logger suppression, it would be nice to have this as a function/method with a wrapper...

Reply:
Can you explain a bit more?

Reply:
there is a single block of code which you want to execute "silently"
co make it as a separate function and write a wrapper which disables the logger and later restore